test/gtest/uct/uct_test.cc

```diff
@@ -1018,6 +1018,48 @@ void uct_test::entity::rkey_release(const uct_rkey_bundle *rkey_bundle) const
     }
 }
 
+void *uct_test::entity::memh_export(uct_mem_h memh) const
+{
+    if (!(md_attr().flags & UCT_MD_FLAG_EXPORTED_MKEY)) {
+        return NULL;
+    }
+
+    void* rkey_buffer                = malloc(md_attr().rkey_packed_size);
+    uct_md_mkey_pack_params_t params = {
+        .field_mask                  = UCT_MD_MKEY_PACK_FIELD_FLAGS,
+        .flags                       = UCT_MD_MKEY_PACK_FLAG_EXPORT
+    };
+
+    if (rkey_buffer == NULL) {
+        UCS_TEST_ABORT("Failed to allocate exported rkey buffer");
+    }
+
+    ucs_status_t status = uct_md_mkey_pack_v2(m_md, memh, NULL, SIZE_MAX,
+                                              &params, rkey_buffer);
+    ASSERT_UCS_OK(status);
+    return rkey_buffer;
+}
+
+void uct_test::entity::memh_attach(void *rkey_buffer,
+                                   uct_rkey_bundle *rkey_bundle)
+{
+    uct_md_mem_attach_params_t attach_params = {0};
+    ucs_status_t status                      =
+            uct_md_mem_attach(m_md, rkey_buffer, &attach_params, &m_exp_memh);
+    ASSERT_UCS_OK(status);
+    free(rkey_buffer);
+
+    void *rkey_repack_buffer = malloc(md_attr().rkey_packed_size);
```

Comment:
I would consider std::array or std::vector here for auto memory handling
It will be also exception safe